kiali/kiali
# Autodiscover istio configmap and deprecate option

### Describe the change

Autodiscovers the istio configmap rather than relying on users to manually specify it. If you have manually specified the config option, then that will be honored for the time being. Moves autodiscovery into the `istio` package.

### Steps to test the PR

1. Deploy istio and kiali
2. Verify that `external_services.istio.config_map_name=""`
3. Verify that you can see the configuration for the istiod on the mesh page

### Automation testing

Unit testing included.

### Issue reference

Relates to https://github.com/kiali/kiali/issues/7177
